### PR TITLE
Clarify cross-chain applications as examples

### LINKED ISSUE #134: Clarify CrossChainApplications as examples
**Context and scope**
The CrossChainApplications directory currently has example applications built on top of Teleporter, for example `ERC20Bridge`. They mostly serve as examples for developers who are looking to build applications on top of Teleporter, and to demonstrate how to use/integrate with Teleporter, but are meant to serve as examples.

**Discussion and alternatives**
Change directory name to include example
Clarify in any relevant documentation

### PR DESCRIPTION
**Pull request by minghinmatthewlam**
## Why this should be merged
Fixes #134 

## How this works
- Updates to clarify in documentation that cross-chain applications are examples
- Adds `@teleporter` in remappings, and used for cross-chain apps
- Update other READMEs, like adding one for example messenger

## How this was tested
CI

## How is this documented
READMEs

### CODE REVIEW

File: contracts/remappings.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
 @subnet-evm-contracts=lib/subnet-evm/contracts/contracts
-@openzeppelin/=lib/openzeppelin-contracts/
\ No newline at end of file
+@openzeppelin/=lib/openzeppelin-contracts/
+@teleporter/=src/Teleporter/
```
Comment (geoff-vball) [inline contracts/remappings.txt:3]: Do we also have to add this to `slither.config`?

**Review by geoff-vball — COMMENTED** [review]
I see there are some formatting changes here, which are good - but it would be helpful if we had some way to enforce uniform formatting as we do now with our `.go` files. I think we can use `forge fmt`, we would just have to add the formatting rules to `foundry.toml`. We could add `forge fmt` to abi bindings script (and probably rename it) and then also check it in ci.

**Comment by michaelkaplan13** [comment]
In each of the example contract `.sol` files, can we add a comment towards the top stating that it is example code and do not use it in production? Something similar to how Chainlink has it for CCIP examples [here](https://github.com/smartcontractkit/ccip-starter-kit-foundry/blob/main/src/BasicMessageReceiver.sol#L8).

File: contracts/remappings.txt (hunk 1 above)
Comment (minghinmatthewlam) [reply contracts/remappings.txt:3]: turns out slither with a foundry project already detects the remappings so no need to. Also removed extra solc remaps we previously had in `slither.confg`

**Comment by minghinmatthewlam** [comment]
> I see there are some formatting changes here, which are good - but it would be helpful if we had some way to enforce uniform formatting as we do now with our `.go` files. I think we can use `forge fmt`, we would just have to add the formatting rules to `foundry.toml`. We could add `forge fmt` to abi bindings script (and probably rename it) and then also check it in ci.

Deferring this to a new issue https://github.com/ava-labs/teleporter/issues/198.

**Review by geoff-vball — DISMISSED** [review]

**Review by geoff-vball — APPROVED** [review]

File: contracts/src/CrossChainApplications/examples/ERC20Bridge/tests/ERC20BridgeTests.t.sol [hunk 2]
```diff
@@ -7,8 +7,8 @@ pragma solidity 0.8.18;
 
 import {Test} from "forge-std/Test.sol";
 import {ERC20Bridge, BridgeToken, IERC20, ERC20, TeleporterMessageInput, TeleporterFeeInfo, IWarpMessenger, ITeleporterMessenger} from "../ERC20Bridge.sol";
-import {TeleporterRegistry} from "../../../Teleporter/upgrades/TeleporterRegistry.sol";
-import {UnitTestMockERC20} from "../../../Mocks/UnitTestMockERC20.sol";
+import {TeleporterRegistry} from "@teleporter/upgrades/TeleporterRegistry.sol";
+import {UnitTestMockERC20} from "../../../../Mocks/UnitTestMockERC20.sol";
```
Comment (cam-schultz) [inline contracts/src/CrossChainApplications/examples/ERC20Bridge/tests/ERC20BridgeTests.t.sol:11]: Could we also add a remapping for the `Mocks` directory?

File: contracts/src/CrossChainApplications/examples/ExampleMessenger/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Example Cross-chain Messenger
+
+An example cross-chain messenger built on top of Teleporter that sends and receives messages between chains.
+
+## Design
+
+- `ExampleCrossChainMessenger`:
+  - `sendMessage` utilizes the Teleporter protocol to start the cross-chain messaging process
+  - `receiveTeleporterMessage` is the handler for receiving Teleporter messages, and saves the most recently sent message from each source chain.
```
Comment (cam-schultz) [inline contracts/src/CrossChainApplications/examples/ExampleMessenger/README.md:9]:
```suggestion
  - `_receiveTeleporterMessage` is the handler for receiving Teleporter messages, and saves the most recently sent message from each source chain.
```

**Review by cam-schultz — COMMENTED** [review]
Two tiny nits, and I agree with Michael's comment about adding comments wherever we can indicating the contracts are not production worthy.

File: contracts/src/CrossChainApplications/examples/NativeTokenBridge/tests/NativeTokenDestinationTests.t.sol [hunk 4]
```diff
@@ -94,6 +102,7 @@ contract NativeTokenDestinationTest is Test {
             _DEFAULT_OTHER_BRIDGE_ADDRESS,
             abi.encode(_DEFAULT_RECIPIENT, _DEFAULT_INITIAL_RESERVE_IMBALANCE)
         );
+        vm.stopPrank();
```
Comment (cam-schultz) [inline contracts/src/CrossChainApplications/examples/NativeTokenBridge/tests/NativeTokenDestinationTests.t.sol:105]: Does `vm.stopPrank` need to be called for every `vm.prank`?

**Review by cam-schultz — APPROVED** [review]
LGTM/. One question about `vm.stopPrank` in the tests.

File: contracts/src/CrossChainApplications/examples/NativeTokenBridge/tests/NativeTokenDestinationTests.t.sol (hunk 4 above)
Comment (minghinmatthewlam) [reply contracts/src/CrossChainApplications/examples/NativeTokenBridge/tests/NativeTokenDestinationTests.t.sol]: no it does not, was not meant to be pushed, thanks for catching

**Review by geoff-vball — APPROVED** [review]

**Review by cam-schultz — APPROVED** [review]

**Pull request merged**